src/Accounts/Accounts/Account/ConnectAzureRmAccount.cs

```diff
@@ -223,6 +222,13 @@ protected override void BeginProcessing()
             AzureSession.Instance.TryGetComponent(WriteWarningKey, out _originalWriteWarning);
             AzureSession.Instance.UnregisterComponent<EventHandler<StreamEventArgs>>(WriteWarningKey);
             AzureSession.Instance.RegisterComponent(WriteWarningKey, () => _writeWarningEvent);
+
+            // todo: ideally cancellation token should be passed to authentication factory as a parameter
+            // however AuthenticationFactory.Authenticate does not support it
+            // so I store it in AzureSession.Instance as a global variable
+            // todo: CancellationTokenSource should be visiable only in cmdlet class
+            // CancellationTokenSource.Token should be passed to other classes
+            AzureSession.Instance.RegisterComponent("LoginCancellationToken", () => new CancellationTokenSource());
```

Comment:
todo: make "LoginCancellationToken" a const
```suggestion
            AzureSession.Instance.RegisterComponent("LoginCancellationToken", () => new CancellationTokenSource());
```